Framework/Kernel/src/MultiFileNameParser.cpp

```diff
@@ -620,11 +596,11 @@ std::vector<std::vector<unsigned int>> generateRange(unsigned int from,
     while (currentRun >= to) {
       if (addRuns) {
         if (runs.empty())
-          runs.push_back(std::vector<unsigned int>(1, currentRun));
+          runs.emplace_back(std::vector<unsigned int>(1, currentRun));
```

Comment:
I think you can replace this and 603 with `runs.emplace_back(1, currentRun)` which potentially saves you a copy